drivers/serial/uart_bt.c

```diff
@@ -0,0 +1,320 @@
+		if (k_is_in_isr() || !atomic_get(&dev_data->bt.enabled)) {
+			LOG_INF("Ring buffer full, discard %c", c);
+			break;
+		}
+
+		k_sleep(K_MSEC(1));
+	}
+
+	/** Don't flush the data until notifications are enabled. */
+	if (atomic_get(&dev_data->bt.enabled)) {
+		/** Delay will allow buffering some characters before transmitting
+		 * data, so more than one byte is transmitted (e.g: when poll_out is
+		 * called inside a for-loop).
+		 */
+		k_work_reschedule(&dev_data->uart.tx_work, K_MSEC(1));
+	}
+}
+
+static int uart_bt_irq_tx_ready(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	if ((ring_buf_space_get(dev_data->uart.tx_ringbuf) > 0) && dev_data->uart.tx_irq_ena) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static void uart_bt_irq_tx_enable(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	dev_data->uart.tx_irq_ena = true;
+
+	if (uart_bt_irq_tx_ready(dev)) {
+		k_work_submit(&dev_data->uart.cb_work);
+	}
+}
+
+static void uart_bt_irq_tx_disable(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	dev_data->uart.tx_irq_ena = false;
+}
+
+static int uart_bt_irq_rx_ready(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	if (!ring_buf_is_empty(dev_data->uart.rx_ringbuf) && dev_data->uart.rx_irq_ena) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static void uart_bt_irq_rx_enable(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	dev_data->uart.rx_irq_ena = true;
+
+	k_work_submit(&dev_data->uart.cb_work);
+}
+
+static void uart_bt_irq_rx_disable(const struct device *dev)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	dev_data->uart.rx_irq_ena = false;
+}
+
+static int uart_bt_irq_is_pending(const struct device *dev)
+{
+	return uart_bt_irq_rx_ready(dev);
+}
+
+static int uart_bt_irq_update(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 1;
+}
+
+static void uart_bt_irq_callback_set(const struct device *dev,
+				     uart_irq_callback_user_data_t cb,
+				     void *cb_data)
+{
+	struct uart_bt_data *dev_data = (struct uart_bt_data *)dev->data;
+
+	dev_data->uart.callback.cb = cb;
+	dev_data->uart.callback.cb_data = cb_data;
+}
+
+static const struct uart_driver_api uart_bt_driver_api = {
+	.poll_in = uart_bt_poll_in,
+	.poll_out = uart_bt_poll_out,
+	.fifo_fill = uart_bt_fifo_fill,
+	.fifo_read = uart_bt_fifo_read,
+	.irq_tx_enable = uart_bt_irq_tx_enable,
+	.irq_tx_disable = uart_bt_irq_tx_disable,
+	.irq_tx_ready = uart_bt_irq_tx_ready,
+	.irq_rx_enable = uart_bt_irq_rx_enable,
+	.irq_rx_disable = uart_bt_irq_rx_disable,
+	.irq_rx_ready = uart_bt_irq_rx_ready,
+	.irq_is_pending = uart_bt_irq_is_pending,
+	.irq_update = uart_bt_irq_update,
+	.irq_callback_set = uart_bt_irq_callback_set,
+};
+
+static int uart_bt_init(const struct device *dev)
+{
+	int err = -1;
```

Comment:
Unnecessary initialization of the variable here (it's always unconditionally set in the function before trying to read it)